code/history.py

```diff
@@ -85,3 +79,11 @@ def history_get(number: int):
         """returns the history entry at the specified index"""
         num = (0 - number) - 1
         return history[num]
+
+    def history_transform_phrase_text(words: list[str]) -> Optional[str]:
+        """Transforms phrase text for presentation in history. Return `None` to omit from history"""
+
+        if not actions.speech.enabled():
+            return None
+
+        return ' '.join(words) if words else None
```

Comment:
The 'if words else None' seems redundant here. I guess the idea is to return None if the list is empty? But the empty string is falsey in python, so the test on line 21 (`if text := ...`) will catch that. Either the test on line 21 should be for `None`, or this `if words else None` should be ditched.